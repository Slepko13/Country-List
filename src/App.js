import React, {useState} from 'react';
import "bootstrap/dist/css/bootstrap.min.css";
import './App.scss';
import CountryList from "./components/CountryList/CountryList";
import CountryDetails from "./components/CoutryDetails/CountryDetails";
import TopBlock from "./components/TopBlock/TopBlock";
import CountryDetailsMobile from "./components/CountryDetailsMobile/CountryDetailsMobile";
import CountryListMobile from "./components/CountryListMobile/CountryListMobile";
import MainHeader from "./components/MainHeader/MainHeader";

function App() {
    const [id, setId] = useState(null);
    const [countryDetails, setCountryDetails] = useState(false);

    return (
        <div className="App ">
            <MainHeader
                countryDetails={countryDetails}
                setCountryDetails={setCountryDetails}
            />
            <CountryDetailsMobile
                id={id}
                countryDetails={countryDetails}
            />
            <div className="content container-fluid ">
                <div className="row ">
                    <div className="col-12 col-lg-8 order-lg-2 ">
                        <TopBlock
                            id={id}
                            countryDetails={countryDetails}
                        />
                        <CountryDetails
                            id={id}
                        />
                    </div>
                    <div className="col-12  col-lg-4">
                        <CountryList
                            setId={setId}
                        />
                        <CountryListMobile
                            setId={setId}
                            countryDetails={countryDetails}
                            setCountryDetails={setCountryDetails}
                        />
                    </div>
                </div>
            </div>
        </div>
    );
}

export default App;
