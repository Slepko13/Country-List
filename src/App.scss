.App {
  display:flex;
  flex-direction: column;
  background: #F8F8F8;

  .content {
    flex:2;
    max-width: 1440px;
  }
}
@media screen and (min-width : 992px){
  .App{
    padding-bottom: 60px;
    .content {
    }
  }
}

