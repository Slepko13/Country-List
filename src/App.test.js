import React from "react";
import App from './App';

describe("App component", ()=>{

    const component = shallow(<App/>);

    it("should renders properly ", () => {
        expect(component).toMatchSnapshot();
    });

    it("should renders properly ", () => {
        expect(component.find('App')).not.toBeNull();
    });

})