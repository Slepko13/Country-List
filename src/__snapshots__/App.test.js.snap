// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList component should renders properly  1`] = `
<div
  className="App "
>
  <MyHeader
    countryDetails={false}
    setCountryDetails={[Function]}
  />
  <CountryDetailsMobile
    countryDetails={false}
    id={null}
  />
  <div
    className="content container-fluid "
  >
    <div
      className="row "
    >
      <div
        className="col-12 col-lg-8 order-lg-2 "
      >
        <TopBlock
          id={null}
        />
        <CountryDetails
          id={null}
        />
      </div>
      <div
        className="col-12  col-lg-4"
      >
        <CountryList
          setId={[Function]}
        />
        <CountryListMobile
          countryDetails={false}
          setCountryDetails={[Function]}
          setId={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
